lib/Controller/SettingsController.php

```diff
@@ -218,4 +225,15 @@ public function GetSettings()
 		];
 		return $result;
 	}
+
+
+	public function SaveSciencemeshSettings()
+	{
+		$sciencemesh_iop_url = $this->request->getParam('sciencemesh_iop_url');
+		$this->serverConfig->setIopUrl($sciencemesh_iop_url);
+		$this->serverConfig->getRevaLoopbackSecret();
```

Comment:
what are these get functions for?

Reply:
This will get/set Reva loop back secret.

Reply:
Ok, I know but you did not fill any variable with their return value?

Reply:
If there're no REVA LOOP BACK or IOP URL set, we will have error cause I'm not sure these two variables were set anywhere else that's why I run these two function to set it first then read it on template file.

Reply:
as I recall, It was a singleton pattern and will create the value and send the result at the first step. 

https://github.com/pondersource/oc-sciencemesh/blob/47d0d19c9f5135852f8e3a1bc8928b4d67b615a1/lib/ServerConfig.php#L55

somewhere you are getting the value and set in view controls. so, at this step, it will be assigned and you do not need to initiate it again.